dpnp/backend/tests/test_random.cpp

```diff
@@ -64,115 +113,75 @@ bool check_statistics(_DataType* r, double tM, double tD, double tQ, size_t size
         return true;
 }
 
-TEST(TestBackendRandomBeta, test_seed)
+TEST_F(RandomTestCase, rng_beta_test_seed)
 {
-    const size_t size = 256;
-    size_t seed = 10;
-    double a = 0.4;
-    double b = 0.5;
-
-    auto QueueOptionsDevices = std::vector<QueueOptions>{QueueOptions::CPU_SELECTOR, QueueOptions::GPU_SELECTOR};
+    const size_t seed = 10;
+    const double a = 0.4;
+    const double b = 0.5;
 
-    for (auto device_selector : QueueOptionsDevices)
-    {
-        dpnp_queue_initialize_c(device_selector);
-        double* result1 = (double*)dpnp_memory_alloc_c(size * sizeof(double));
-        double* result2 = (double*)dpnp_memory_alloc_c(size * sizeof(double));
-
-        dpnp_rng_srand_c(seed);
-        dpnp_rng_beta_c<double>(result1, a, b, size);
+    dpnp_rng_srand_c(seed);
+    dpnp_rng_beta_c<double>(result1, a, b, size);
 
-        dpnp_rng_srand_c(seed);
-        dpnp_rng_beta_c<double>(result2, a, b, size);
+    dpnp_rng_srand_c(seed);
+    dpnp_rng_beta_c<double>(result2, a, b, size);
 
-        for (size_t i = 0; i < size; ++i)
-        {
-            EXPECT_NEAR(result1[i], result2[i], 0.004);
-        }
+    for (size_t i = 0; i < size; ++i)
+    {
+        EXPECT_NEAR(result1[i], result2[i], 0.004);
```

Comment:
why you are expecting close but not the same values?
looks like values should be the same if you are using the same seed.

Reply:
it is common case for checking floating point values. But you are right it is better check the same values here for the same device calculation.

Reply:
@Alexander-Makaryev this changes will be done on refactoring this common tests on other PR. This tests case will be generalized for most random distribution functions.  Needed changes will be done there. I am leaving this PR as is.